### PR TITLE
Improve testing

### PR DESCRIPTION
**Pull request by victorvarza**
## Description

Testing refactoring.

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -163,15 +179,15 @@ generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and
...
-GOBIN=$(shell pwd)/bin go get $(2) ;\
-rm -rf $$TMP_DIR ;\
-}
+	@[ -f $(1) ] || {						\
+	set -e ;								\
+	TMP_DIR=$$(mktemp -d) ;					\
+	cd $$TMP_DIR ;							\
+	go mod init tmp ;						\
+	echo "Downloading $(2)";				\
+	GOBIN=$(shell pwd)/bin go get $(2);	\
```
Comment (palasanu) [inline Makefile:188]: The '\' here should be at the same indentation as the others

File: cmd/cc/client.go [hunk 2]
```diff
@@ -120,7 +120,13 @@ func main() {
 		os.Exit(1)
 	}
 
-	appConfig := utils.LoadClientConfig()
+	appConfig, err := utils.LoadClientConfig()
+
+	if err != nil {
+		setupLog.Error(err, "failed to load client configuration")
+		os.Exit(1)
```
Comment (palasanu) [inline cmd/cc/client.go:127]: Is there a reason for not using `log.Fatalf()` like in `cmd/api/api.go` at line 45?

>  appConfig := utils.LoadApiConfig()
> 	appConfig, err := utils.LoadApiConfig()
> 	if err != nil {
> 		log.Fatalf("Cannot load the api configuration: '%v'", err.Error())
> 	}

**Review by palasanu — APPROVED** [review]
LGTM

File: cmd/cc/client.go (hunk 2 above)
Comment (victorvarza) [reply cmd/cc/client.go:127]: good catch, thanks

Comment (victorvarza) [reply cmd/cc/client.go:127]: Actually setupLog does more stuff in the kubebuilder client for logging config. I'll keep it as it is for client codebase consistency.

**Pull request merged**